learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+
+```{r}
+#| label: "load-readmission"
+#| eval: false
+library(readmission)
+
+readmission
+```
+
+The first variable in this data, `readmitted`, gives whether the patient was readmitted within 30 days of discharge. We'll use this variable as a proxy for "unmet need for additional care," in that readmission within one month indicates that the patient may have benefited from additional attention during their hospital stay; if a machine learning model consistently identifies lesser need (via prediction of non-readmission) in one subgroup than another, the subgroup allocated lesser need will go without care they'd benefit from. We'd like to train a model that is both fair with regard to how it treats `race` groups and is as performant as possible. The tidymodels framework provides the tools needed to identify such disparities.
+
+Loading needed packages:
+
+```{r}
+#| label: "load-tidymodels"
+library(tidymodels)
+library(baguette)
+```
+
+## Exploratory Analysis
+
+Let's start off our analysis with some explanatory summarization and plotting. First, taking a look at our outcome variable:
+
+```{r}
+#| label: "count-by-readmitted"
+readmission %>%
+  count(readmitted)
+```
+
+`r round(mean(readmission$readmitted == "Yes"), 3)*100`% of patients readmitted within 30 days after being discharged from the hospital. This is an example of a modeling problem with a _class imbalance_, where one value of the outcome variable is much more common than another. Now, taking a look at the counts of those in each protected class:
+
+```{r}
+#| label: "count-by-race"
+readmission %>%
+  count(race)
+```
+
+::: callout-note
+We'll refer to the `race` groups in this data by their actual value (e.g. `"Caucasian"` rather than Caucasian) so as to not to take for granted the choices that the dataset authors made in choosing these categorizations. Racial categorizations are not stable across time and place---as you read on, consider how a change in the categories used in data collection might affect this analysis [@omi1994].
+:::
+
+A vast majority of patients are labeled `"Caucasian"` (`r round(mean(readmission$race == "Caucasian"), 3)*100`%) or `"African American"` (`r round(mean(readmission$race == "African American"), 3)*100`%). The counts for the remaining racial categorizations are quite a bit smaller, and when we split the data up into resamples, those counts will reduce even further. As a result, the variability associated with the estimates `"Asian"`, `"Hispanic"`, `"Other"`, and `"Unknown"` will be larger than those for `"African American"` and `"Causasian"`. As an example:
+
+```{r}
+#| label: "prop-by-race"
+#| message: false
+#| warning: false
+readmission %>%
```

Comment:
I think that standard errors here are not great. They are complete driven by the magnitude of the proportions. Instead, I'd use a combination of `prop.test()` and `tidy()` to get the confidence intervals back.